Excel_Adapter/Validation/WorksheetName.cs

```diff
@@ -0,0 +1,209 @@
+    public partial class Validation
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static string WorksheetName(string name, IXLWorkbook workbook)
+        {
+            string worksheetName = name;
+
+            List<WorksheetValidation> validationIssues = new List<WorksheetValidation>();
+
+            bool isValidName = IsValidName(worksheetName, workbook, out validationIssues);
+            while (!isValidName)
+            {
+                worksheetName = ModifyWorksheetName(worksheetName, validationIssues, workbook);
+                isValidName = IsValidName(worksheetName, workbook, out validationIssues);
+            }
+
+
+            if (worksheetName != name)
+            {
+                BH.Engine.Base.Compute.RecordError("Name of worksheet has been adjusted to a name that which is compatible with Excel naming limitations.");
+            }
+
+            return worksheetName;
+        }
+
+
+        /***************************************************/
+
+        private static string ModifyWorksheetName(string worksheetName, List<WorksheetValidation> validationIssues, IXLWorkbook workbook)
+        {
+            switch (validationIssues[0])
+            {
+                case WorksheetValidation.isReservedWord:
+                case WorksheetValidation.isBlank:
+                    return SetGeneralName();
+
+                case WorksheetValidation.isNotUnique:
+                    return SetUniqueName(workbook, worksheetName);
+
+                case WorksheetValidation.isBeginOrEndWithInvalidCharacter:
+                    return RemoveInvalidBeginningOrEnding(worksheetName);
+
+                case WorksheetValidation.isNotValidCharacters:
+                    return RemoveInvalidCharacters(worksheetName);
+
+                case WorksheetValidation.isNotWithinCharacterLimit:
+                    return TrimName(worksheetName);
+
+                default:
+                    break;
+            }
+
+            return worksheetName;
+        }
+
+        private static string TrimName(string worksheetName)
+        {
+            if (worksheetName.ToString().Contains(" "))
+            {
+                worksheetName = worksheetName.Replace(" ", "");
+                return worksheetName;
+            }
+
+            return worksheetName.Substring(0, 31);
+
+        }
+
+        private static string RemoveInvalidCharacters(string worksheetName)
+        {
+            List<char> InvalidChars = new List<char>() { '[', '/', '?', ']', '*', '\\', ':' };
+
+            InvalidChars.ForEach(c => worksheetName = worksheetName.Replace(c.ToString(), String.Empty));
+            return worksheetName;
+        }
+
+        private static string RemoveInvalidBeginningOrEnding(string worksheetName)
+        {
+            if (worksheetName.Substring(0, 1) == "\'")
```

Comment:
```suggestion
            if (worksheetName.StartsWith("\'"))
```

Alternatively

```suggestion
            if (worksheetName[0] == '\'')
```

But this one is less intuitive than `StartsWith`